### DISCUSSION TITLE
recover from message failed to send?

### DISCUSSION THREAD

**Discussion by csaftoiu** [eternnoir/pyTelegramBotAPI#1690] [General]
Sometimes messages are failing to send. Restarting the script recovers it. But the messages that I asked to send with `bot.send_message` don't get re-sent. Is there any built-in way to have it recover this on startup or do I have to roll my own?

**Comment by Badiboy** [comment]
roll my own.

But restarting script is a strange way. Why not just to retry sending message?
Or catch exception and check?
Or wonder in debug mode (logger.DEBUG) and see what happen there...

**Comment by csaftoiu** [comment]
Actually I see it does have built-in retry already ! https://github.com/eternnoir/pyTelegramBotAPI/blob/c8d1dac61eaff9312ef450bac790e37e398e0740/telebot/apihelper.py#L39 

Restarting the script -- I knew it worked, but I think simpler to just retry, it doesn't seem script restart is necessary
